### ISSUE TITLE
"nix-shell -p qt5.base" fails on 2nd run

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#10273]
... due to left-over files created the first run:

```
$ mkdir /tmp/test
$ cd /tmp/test
$ nix-shell -p qt5.base
qmake: File exists
[nix-shell:/tmp/test]$ tree -L 2
.
└── qmake-shell
    ├── bin
    ├── include
    ├── lib
    ├── mkspecs
    ├── nix-support
    └── share

7 directories, 0 files
[nix-shell:/tmp/test]$ exit
$ nix-shell -p qt5.base
cp: cannot create regular file ‘/tmp/test/qmake-shell/bin/qmake’: Permission denied
$ 
```

**Comment by ttuegel** [comment]
We should use `mktemp -d` to create the temporary directory instead of giving it a deterministic name.

**Comment by ttuegel** [comment]
As reported by @Profpatsch, this continues to be an issue.

**Comment by jagajaga** [comment]
Any updates?

**Comment by ttuegel** [comment]
I can't reproduce this with a recent master. @Profpatsch ?

**Comment by Profpatsch** [comment]
@ttuegel I’m baffled, because there doesn’t seem to be a `qt5.base` anymore. oO

**Comment by RocketPuppy** [comment]
I'm seeing this with qt5.webkit. Should this be reopened?

**Comment by ttuegel** [comment]
It's fixed in unstable, but I can't backport the fix to 15.09 based on our policies. Luckily, 16.03 will come out soon.
